### PR TITLE
add LAA-HMRC interface architecture

### PR DESCRIPTION
**Pull request by willc-work**
## What does this pull request do?

Add the new LAA-HMRC interface to the Apply for Legal Aid ad LAA architecture

## What is the intent behind these changes?
keep the architecture diagrams up to date

### CODE REVIEW

File: src/main/kotlin/model/HMRCInterface.kt [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+class HMRCInterface private constructor() {
+  companion object : LAASoftwareSystem {
+    lateinit var system: SoftwareSystem
+    lateinit var api: Container
+
+    override fun defineModelEntities(model: Model) {
+      system = model.addSoftwareSystem(
+        "LAA-HMRC Interface Service API",
+        "A service to facilitate HMRC API access for the LAA Use Cases"
```
Comment (oliverweighell) [inline src/main/kotlin/model/HMRCInterface.kt]: LAA Use Cases might be a bit oblique for those not involved in the HMRC work. How about something like 'An interface with HMRC providing access to income, fraud and debt data'?

File: src/main/kotlin/model/HMRCInterface.kt [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+      ).apply {
+        Tags.GET_LEGAL_AID.addTo(this)
+      }
+
+      api = system.addContainer(
+        "LAA-HMRC Interface Service API",
+        "This Service facilitates the HMRC API access for the LAA Use Cases",
```
Comment (oliverweighell) [inline src/main/kotlin/model/HMRCInterface.kt]: As per the above point, how about 'A JSON interface for the HMRC API'

**Review by oliverweighell — CHANGES_REQUESTED** [review]
We should add HMRC as an external dependency. A good example of this is Benefit checker and DWP: 
- Relationship: https://github.com/ministryofjustice/laa-architecture-as-code/blob/main/src/main/kotlin/model/BenefitChecker.kt#L35-L43
- DWP: https://github.com/ministryofjustice/laa-architecture-as-code/blob/main/src/main/kotlin/model/DWP.kt

Note - you'll also need to add HMRC in the systems list here https://github.com/ministryofjustice/laa-architecture-as-code/blob/main/src/main/kotlin/defineWorkspace.kt#L10

File: src/main/kotlin/model/HMRCInterface.kt [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+    override fun defineInternalContainerRelationships() {
+    }
+
+    override fun defineRelationships() {
+    }
+
+    override fun defineExternalRelationships() {
+      api.uses(
+        HMRC.hmrcInterface,
```
Comment (willc-work) [inline src/main/kotlin/model/HMRCInterface.kt:39]: I have just called this hmrcInterface but perhaps employmentChecker or financesChecker would be more appropriate. Let me know what you think and I will update it.

**Review by oliverweighell — APPROVED** [review]

**Pull request merged**